pkg/apis/ibmcloudprovider/v1beta1/ibmcloudproviderconfig_types.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// IBMCloudMachineProviderSpec is the type that will be embedded in a Machine.Spec.ProviderSpec field
+// for an IBM Cloud VPC virtual machine. It is used by the IBM Cloud machine actuator to create a single Machine.
+// +k8s:openapi-gen=true
+type IBMCloudMachineProviderSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// vpc type Instance struct
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The user-defined name for this virtual server instance
+	// default: system hostname
+	Name string `json:"name,omitempty"`
+
+	// VPC name where the instance will be created
+	VPC string `json:"vpc"`
+
+	// Actuator will apply these tags to an virtual server instance if not present in additon
+	// to default tags applied by the actuator
+	// TODO
+	// Tags []string `json:"tags,omitempty"`
+
+	// TODO: Add labels to the virtual server
+	// Labels map[string]string `json:"labels,omitempty"`
+
+	// Image is the id of the custom OS image in VPC
+	// Example: a123-ade15224-ed7y-5e44-ye56-8199d4557699
+	Image string `json:"image"`
```

Comment:
I suggest we use the Image name instead of ID. Incoming changes in the installer to support automated custom image uploading will necessitate this since the ID will be generated after a Terraform apply.